tools/autograd/derivatives.yaml

```diff
@@ -983,7 +983,7 @@
 - name: maximum(Tensor self, Tensor other) -> Tensor
   self: at::where(self == other, grad / 2, grad).masked_fill_(self < other, 0)
   other: at::where(self == other, grad / 2, grad).masked_fill_(self > other, 0)
-  result: other_t + at::where(self_p == other_p, 0.5, (self_p > other_p).to(result.scalar_type())) * (self_t - other_t)
```

Comment:
Does this work? From `where`'s docs, it looks like it does not support float32 as the dtype for the tensor. Maybe we could use masked_fill instead?

> Currently valid scalar and tensor combination are 1. Scalar of floating dtype and torch.double 2. Scalar of integral dtype and torch.long 3. Scalar of complex dtype and torch.complex128

https://pytorch.org/docs/stable/generated/torch.where.html#torch.where

Reply:
Yes, the tests pass locally. Below is some visual evidence that torch.where supports tensors with float32 dtype.

![image](https://user-images.githubusercontent.com/5652049/161861588-4f525869-63e9-4100-b4b1-92ff20ec74a2.png)

The documentation is saying "IF you pass in a Python Number (Scalar) that is floating point, THEN any tensors need to be torch.double". Here's some evidence for that:
![image](https://user-images.githubusercontent.com/5652049/161861741-fb260582-898c-4115-9d91-91d7da810d73.png)